### DISCUSSION TITLE
How to set default values for Socials in Social Block

### DISCUSSION THREAD

**Discussion by kurund** [voidlabs/mosaico#704] [Q&A] [answered]
I have created a new template using default  `versafix-1` template and I want to set the default values for socials like FB etc. Is this possible?

**Comment by bago** [comment]
Defaults are usually read from the html: have you tried adding your defaults to the master template html?

**Reply by kurund** [reply]
Thanks for your response.

```bash
      socialBlock {
        label: Social Block;
        properties: socialIconType=colors fbVisible=true fbUrl twVisible=true twUrl ggVisible=false ggUrl webVisible=true webUrl waVisible=false waUrl tgVisible=false tgUrl inVisible=false inUrl piVisible=false piUrl flVisible=false flUrl viVisible=false viUrl instVisible=true instUrl youVisible=true youUrl ttVisible=true ttUrl tcVisible=false tcUrl longTextStyle longText backgroundColor;
        variant:socialIconType;
        theme: frameTheme
      }
```
I was looking above code to set defaults. For eg: I can change the defaults for toggle fields `fbVisible=false` to `fbVisible=true`. However, I can't set the default url for `fbUrl`. I think it inherits url widget and not sure we can set the default for it.

I can change the default for `longText` by directly editing the html.

My use case is setting default facebook and twitter url so that we don't need to update it for every new mailings manually. 

Am I missing something obvious?

**Reply by bago** [reply]
the defaults are written there only for some specific value that cannot be read from the HTML.
Just put your values in the HTML.

e.g, this is the div for facebook icon:
`
              <div data-ko-wrap="false" style="display: inline-block;" data-ko-display="fbVisible">&nbsp;<a href="" data-ko-display="fbVisible" style="border-radius: 50px; -ko-attr-href: @[fbUrl]"><img border="0" style="display: inline-block; vertical-align: top; padding-bottom: 0px; -ko-attr-src: @[Url.templatePath('img/icons/fb-'+bigSocialIconType+'-96.png')]; -ko-attr-width: @[bigSocialIconSize]; -ko-attr-height: @[bigSocialIconSize]" src="img/icons/fb-rdcol-96.png" width="48" height="48" alt="Facebook"></a></div>`

just put your url in the href="" attribute and it should be automatically taken as the default value.

**Reply by kurund** [reply]
thank you. I did try that. However, when I do so then the entire editor breaks.

These are my changes:
```html
                <div data-ko-wrap="false" style="display: inline-block;" data-ko-display="fbVisible">&nbsp;<a href="https://facebook.com/mypage" data-ko-display="fbVisible" style="border-radius: 50px; -ko-attr-href: @[fbUrl]"><img border="0" style="display: inline-block; vertical-align: top; padding-bottom: 0px; -ko-attr-src: @[Url.templatePath('img/icons/fb-'+socialIconType+'-96.png')]" src="img/icons/fb-rdcol-96.png" width="32" height="32" alt="Facebook"></a></div>
```
Here is the error in console.

![Screenshot from 2024-01-11 14-07-51](https://github.com/voidlabs/mosaico/assets/151481/4cfd9499-af36-465d-96a4-a044bf109455)

**Reply by bago** [reply, marked as answer]
If you change that in the "socialBlock" maybe there are 2 (or more) occourences of fbUrl because of alternative versions of the same block. You have to put the same href for all of them (in the same block).

Also, make sure you didn't left anything weird in the "properties:" option in socialBlock definition when you attempted to add a default value there.

**Reply by kurund** [reply]
Thank you so much. 

That was it. I had to set the same `href` for all the instance of `fbUrl`

Really appreciate your help!
